src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/PartialTransactionRequester.cs

```diff
@@ -0,0 +1,110 @@
+            ILoggerFactory loggerFactory,
+            ICrossChainTransferStore crossChainTransferStore,
+            IAsyncLoopFactory asyncLoopFactory,
+            INodeLifetime nodeLifetime,
+            IConnectionManager connectionManager,
+            IFederationGatewaySettings federationGatewaySettings)
+        {
+            Guard.NotNull(loggerFactory, nameof(loggerFactory));
+            Guard.NotNull(crossChainTransferStore, nameof(crossChainTransferStore));
+            Guard.NotNull(asyncLoopFactory, nameof(asyncLoopFactory));
+            Guard.NotNull(nodeLifetime, nameof(nodeLifetime));
+            Guard.NotNull(federationGatewaySettings, nameof(federationGatewaySettings));
+
+            this.loggerFactory = loggerFactory;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.crossChainTransferStore = crossChainTransferStore;
+            this.asyncLoopFactory = asyncLoopFactory;
+            this.nodeLifetime = nodeLifetime;
+            this.connectionManager = connectionManager;
+            this.federationGatewaySettings = federationGatewaySettings;
+        }
+
+        /// <summary>
+        /// Broadcast the partial transaction request to federation members.
+        /// </summary>
+        /// <param name="payload">The payload to broadcast.</param>
+        async Task BroadcastAsync(RequestPartialTransactionPayload payload)
+        {
+            this.logger.LogTrace("({0}:'{1}',{2}:'{3}')", nameof(payload.Command), payload.Command, nameof(payload.DepositId), payload.DepositId);
+
+            List<INetworkPeer> peers = this.connectionManager.ConnectedPeers.ToList();
+
+            var ipAddressComparer = new IPAddressComparer();
+
+            foreach (INetworkPeer peer in peers)
+            {
+                if (this.federationGatewaySettings.FederationNodeIpEndPoints.Any(e => ipAddressComparer.Equals(e.Address, peer.PeerEndPoint.Address)))
+                {
+                    try
+                    {
+                        await peer.SendMessageAsync(payload).ConfigureAwait(false);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                    }
+                }
+            }
+
+            this.logger.LogTrace("(-)");
+        }
+
+        public void Start()
+        {
+            this.asyncLoop = this.asyncLoopFactory.Run("Get partial templates job", token =>
+            {
+                Dictionary<uint256, Transaction> transactions = this.crossChainTransferStore.GetTransactionsByStatusAsync(
+                    CrossChainTransferStatus.Partial).GetAwaiter().GetResult();
+
+                foreach (KeyValuePair<uint256, Transaction> kv in transactions)
+                {
+                    BroadcastAsync(new RequestPartialTransactionPayload(kv.Key).AddPartial(kv.Value)).GetAwaiter().GetResult();
+                }
+
+                if (transactions.Count > 0)
```

Comment:
Maybe copy this above the foreach, then change this logger msg to "Partial templates completed" (or broadcasted).

Reply:
The purpose of the message is to indicate when the templates have been requested - which is what the broadcast does...